### PR TITLE
Hawkular Alerting Group Triggers module

### PR DESCRIPTION
**Pull request by dkorn**
Documentation and unittests will be added separately.

SSL certificate verification is not yet supported.

### CODE REVIEW

**Comment by dkorn** [comment]
@yaacov @enoodle @cben @kwoodson please review

File: library/hawkular_alerts_group.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+    severity = module.params['severity']
+    scheme   = module.params['scheme']
+    state    = module.params['state']
+    enabled  = module.params['enabled']
+
+    # TODO: remove once the ssl cert verification is checked
+    import ssl
+    context = ssl._create_unverified_context()
+
+    hawkular_alerts = HawkularAlertsGroupTrigger(module, tenant, hostname, port, scheme, token, context)
```
Comment (simon3z) [inline library/hawkular_alerts_group.py]: Maybe this is shorter:

```python
hawkular_alerts = HawkularAlertsGroupTrigger(
    module,
    module.params['tenant'],
    ...
)
```

Comment (dkorn) [reply library/hawkular_alerts_group.py]: @simon3z I can do that but I tried to maintain consistency with all the previous modules

Comment (simon3z) [reply library/hawkular_alerts_group.py]: :+1:

File: library/hawkular_alerts_group.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+  hawkular_api_auth_token:
+    description:
+      - the hawkular API auth token
+    default: HAWKULAR_TOKEN env var if set, otherwise it is required to pass it
+    required: True
+  tenant:
+    description:
+      - the hawkular tenant
+    default: null
+    required: True
```
Comment (cben) [inline library/hawkular_alerts_group.py]: shouldn't have default & required.  also `group_id` below.

File: library/hawkular_alerts_group.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+  name:
+    description:
+      - the group trigger name
+    default: null
+    required: False
+  group_id:
+    description:
+      - the group trigger id
```
Comment (cben) [inline library/hawkular_alerts_group.py]: Please document (here or elsewhere) that group_id is the "primary field" on which you match existing trigger.
(is there a convention in ansible documentation how/where to word this?)

It's good to always clarify this, but especially here both group_id and name could be reasonable behaviors.

File: library/hawkular_alerts_group.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+            Returns:
+                whether or not a change took place and a short message
+                describing the operation executed
+        """
+        try:
+            self.client.get_trigger(group_id)
+            return dict(
+                msg="Group trigger {group_id} already exist".format(group_id=group_id),
+                changed=self.changed)
```
Comment (cben) [inline library/hawkular_alerts_group.py:110]: What if it exists but has different name, severity, etc.?
If you prefer to do it later, please open an issue.

**Review by cben — COMMENTED** [review]
Generally looks good.

File: library/hawkular_alerts_group.py [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+    tenant   = module.params['tenant']
+    name     = module.params['name']
+    group_id = module.params['group_id']
+    severity = module.params['severity']
+    scheme   = module.params['scheme']
+    state    = module.params['state']
+    enabled  = module.params['enabled']
+
+    # TODO: remove once the ssl cert verification is checked
+    import ssl
```
Comment (cben) [inline library/hawkular_alerts_group.py]: nitpick: there is no strong reason to import here, I'd import at the top (in fact you have a comment there probably referring to this).

File: library/hawkular_alerts_group.py (hunk 2 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: Done, thanks

File: library/hawkular_alerts_group.py (hunk 5 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: Done, thanks

File: library/hawkular_alerts_group.py (hunk 4 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py:110]: in that case we'll update the group trigger.
I added this functionality in https://github.com/dkorn/hawkular-alerts-module/pull/6
which depends on https://github.com/hawkular/hawkular-client-python/pull/28

File: library/hawkular_alerts_group.py (hunk 3 above)
Comment (dkorn) [reply library/hawkular_alerts_group.py]: Added it here, thanks.
I haven't seen a convention but will check for one.

File: library/hawkular_alerts_group.py (hunk 4 above)
Comment (cben) [reply library/hawkular_alerts_group.py:110]: Ah, I see.  LGTM on this part then, reviewing #6 now...

**Review by cben — APPROVED** [review]

**Pull request merged**